packages/web-components/@sfx/search-box/test/unit/search-box.test.ts

```diff
@@ -0,0 +1,208 @@
+  });
+
+  afterEach(() => {
+    searchboxDispatchEvent.restore();
+  });
+
+  describe('Constructor', () => {
+    it('should extend Base', () => {
+      expect(searchbox).to.be.an.instanceof(Base);
+    });
+
+    describe('placeholder property', () => {
+      it('should have default value `Type your search`', () => {
+        expect(searchbox.placeholder).to.equal('Type your search');
+      });
+    });
+
+    describe('value property', () => {
+      it('should have default value of an empty string', () => {
+        expect(searchbox.value).to.equal('');
+      });
+    });
+
+    describe('searchButton property', () => {
+      it('should have default value of false', () => {
+        expect(searchbox.searchButton).to.equal(false);
+      });
+    });
+
+    describe('clearButton property', () => {
+      it('should have default value of false', () => {
+        expect(searchbox.clearButton).to.equal(false);
+      });
+    });
+  });
+
+  describe('connectCallback', () => {
+    it('should add an autocomplete_hover eventListener to the window', () => {
+      const windowAddEventListener = stub(window, 'addEventListener');
+
+      searchbox.connectedCallback();
+
+      expect(windowAddEventListener).to.have.been.calledWith(SEARCHBOX_EVENT.UPDATE_SEARCH_TERM, searchbox.updateText);
+    });
+  });
+
+  describe('disconnectCallback', () => {
+    it('should remove the autocomplete_hover eventListener from the window', () => {
+      const windowRemoveEventListener = stub(window, 'removeEventListener');
+
+      searchbox.disconnectedCallback();
+
+      expect(windowRemoveEventListener).to.have.been.calledWith(
+        SEARCHBOX_EVENT.UPDATE_SEARCH_TERM,
+        searchbox.updateText
+      );
+    });
+  });
+
+  describe('emitSearchEvent', () => {
+    it('should dispatch a search request event', () => {
+      const searchRequestEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCH_REQUEST, {
+        detail: 'a',
+        bubbles: true
+      });
+      searchbox.value = 'a';
+
+      searchbox.emitSearchEvent();
+
+      expect(searchboxDispatchEvent).to.have.been.calledWith(searchRequestEvent);
+    });
+  });
+
+  describe('emitSearchBoxClearedEvent', () => {
+    it('should dispatch an emitSearchBoxClearedEvent', () => {
+      const searchboxClearedEvent = new CustomEvent('sfx::search_box_cleared');
+
+      searchbox.emitSearchBoxClearedEvent();
+
+      expect(searchboxDispatchEvent).to.have.been.calledWith(searchboxClearedEvent);
+    });
+  });
+
+  describe('updateText', () => {
+    it('should update the value property in response to data received', () => {
+      const detail = 'inputText';
+      stub(searchbox, 'getInputElement').returns(
+        html`
+          <input type="text" data-sfx-ref="searchInput" placeholder="Type your search" />
+        `
+      );
+      searchbox.updateText({ detail });
+      expect(searchbox.value).to.equal(detail);
+    });
+  });
```

Comment:
Another test that you want is to make sure that the input element's `value` property is set to the search term. You'll want to stub `getInputElement` to return an empty object (or an object with a `value` property that is different than the expected value) and later expect that that object's `value` property contains the expected search term.

Reply:
I rolled this into the above test, I think that it helps solve the problem that we were seeing. Let me know what you think of the changes.

Reply:
I'm split on whether it should be a separate test or if this is okay, but if you're going to keep this a single test, then the second assertion should be directly asserting that `inputObject.value` is the expected value and not doing a roundabout assertion.